action-server/src/threads/worker.ts

```diff
@@ -1,23 +1,39 @@
 import { threadId } from "worker_threads";
 import { jsExecute } from "../utils/codeRunner";
 import { ActionResponse, ActionTask } from "../type/types";
-import { ActionsDbManager } from "../db";
 import logger from "../utils/logger";
+import {configuration} from "../config";
+import pg from "pg";
+
+const { AERIE_DB, AERIE_DB_HOST, AERIE_DB_PORT, ACTION_DB_USER, ACTION_DB_PASSWORD } = configuration();
+
+function getWorkerDbPool() {
+  // we currently have no way to pass DB connections from a parent-managed db pool to the worker process (preferred).
+  // (Pools/clients contain data that cannot be serialized, causing a DataCloneError if passed to the worker)
+  // instead, when an action is run, a worker creates a new Pool with one client, to get its own connection,
+  // and closes it again when the action run is complete.
+  return new pg.Pool({
+    host: AERIE_DB_HOST,
+    port: parseInt(AERIE_DB_PORT, 5432),
+    database: AERIE_DB,
+    user: ACTION_DB_USER,
+    password: ACTION_DB_PASSWORD,
+    // should have exactly one client/connection
+    min: 1,
+    max: 1
+  });
+}
```

Comment:
I was thinking a little about whether we should keep the database pool alive in the worker threads, as a singleton, to maintain a constant connection. I'm leaning towards your approach of creating a new pool per action run. This releases connections when they're not needed, which is could happen if the user isn't using any actions for a while and the action server is just idle.

Reply:
Yeah that's an interesting idea, the implications aren't completely obvious to me in terms of how it would behave so I think it's best left for later - we're coming back to the pooling question in V2 so I will do some testing with that approach.